### DISCUSSION TITLE
Nfcapd - Sample Configuration File?

### DISCUSSION THREAD

**Discussion by David00** [phaag/nfdump#524] [nfdump-1.7]
Hello,

I've built nfcapd from source in Docker, but I'm having a hard time starting it with a config file. 

I can start the application by specifying the options as flags on the command line, but as soon as I put those options into a config file and pass the `-C /path/to/config` to nfcapd, it doesn't run.  

Here are the contents of my config file, followed by the attempt at starting `nfcapd` with the config file, followed by providing the arguments manually as flags:

```bash
/ # nfcapd -V
nfcapd: Version: 1.7.4-release  Date: Sat Feb 17 15:05:20 CET 2024
/ #
/ # cat nfcapd.config
flowdir=/var/nfdump/flow-data
portnum=9995
bufflen=250000
num=2
/ #
/ # nfcapd -C /nfcapd.config
Failed to parse config file /nfcapd.config: line 1: extra chars after value
/ #
/ # nfcapd -M /var/nfdump/flow-data -p 9995 -B 250000 -S 2
Bound to IPv4 host/IP: any, Port: 9995
Standard setsockopt, SO_RCVBUF is 212992 Requested length is 250000 bytes
System set setsockopt, SO_RCVBUF to 425984 bytes
Init v5/v7: Default sampling: 1
Init v9: Max number of v9 tags enabled: 112, default sampling: 1
Init IPFIX: Max number of ipfix tags enabled: 104, default sampling: 1
Startup nfcapd.
```

I've tried putting quotes around the path, but then nfcapd reports:

```
/ # cat nfcapd.config
flowdir="/var/nfdump/flow-data"
portnum=9995
bufflen=250000
num=2
/ #
/ # nfcapd -C /nfcapd.config
ERROR, No source configurations found
```

I don't want to hardcode the nfcapd configuration by using CLI flags in my Docker image because changing them will require rebuilding the image.  My intent is to map the config file to the host system.

What am I doing wrong?

**Comment by phaag** [comment]
Currently not all cli arguments are available through the config file. See the `nfdump.conf.dist` file in the `src/libnffile/conf` folder. However, it could easily be extended. I can put that on the todo list. You may also open an issue as an enhancement request.
